app/src/main/java/com/architectcoders/aacboard/ui/fragments/NewDashboardFragment.kt

```diff
@@ -1,6 +1,107 @@
 package com.architectcoders.aacboard.ui.fragments
 
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.core.view.isVisible
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
 import com.architectcoders.aacboard.R
+import com.architectcoders.aacboard.databinding.FragmentNewDashboardBinding
+import com.architectcoders.aacboard.domain.data.dashboard.DashboardWithCells
+import com.architectcoders.aacboard.ui.fragments.stateholder.NewDashBoardState
+import com.architectcoders.aacboard.ui.fragments.stateholder.buildNewDashBoardCellState
+import com.architectcoders.aacboard.ui.fragments.viewmodel.NewDashBoardViewModel
+import com.architectcoders.aacboard.ui.utils.loadUrl
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.launch
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
-class NewDashboardFragment : Fragment(R.layout.fragment_new_dashboard)
\ No newline at end of file
+class NewDashboardFragment : Fragment() {
+
+    private var _binding: FragmentNewDashboardBinding? = null
+    private val binding get() = _binding!!
+
+    private val viewModel: NewDashBoardViewModel by viewModel()
+
+    private lateinit var newDashBoardState: NewDashBoardState
+    private var dashBoardId: Int = 0
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        _binding = FragmentNewDashboardBinding.inflate(inflater, container, false)
+        newDashBoardState = buildNewDashBoardCellState()
+        initView()
+        checkSearchResponse()
+        return binding.root
+    }
+
+    private fun initView() {
+        setOnClickListeners()
+        dashBoardId = viewModel.dashBoardId
+        lifecycleScope.launch {
+            viewModel.state.collectLatest { uiState ->
+                binding.progressCircular.isVisible = uiState.isLoading
+                binding.buttonNewDashboardSave.isEnabled = !uiState.isLoading
+                uiState.pictogram?.let {
+                    binding.ivNewDashboardImage.loadUrl(it.url)
+                }
+            }
+        }
+    }
+
+    private fun setOnClickListeners() {
+        binding.buttonNewDashboardCancel.setOnClickListener {
+            findNavController().popBackStack()
```

Comment:
shouldn't this navigation be handled by the state holder through the `onCancel` method??

Reply:
You're right! I completely forgot! Changed.